backend/preprocess.py

```diff
@@ -1,19 +1,101 @@
 
 class Preprocessor:
-    @staticmethod
-    def preprocess_files(zip_file_path: str):
-        extract_dir = 'extracted_repo'
-        os.makedirs(extract_dir, exist_ok=True)
+    def camel_case_split(identifier):
+        """
+        Splits camelCase words via regular expression
 
-        with zipfile.ZipFile(zip_file_path, 'r') as zip_ref:
-            zip_ref.extractall(extract_dir)
+        Args:
+            identifier (string): token to be matched
 
-        files_list = []
-        for dirpath, dirnames, filenames in os.walk(extract_dir):
-            for filename in filenames:
-                file_path = os.path.join(dirpath, filename)
-                files_list.append(file_path)
+        Returns:
+            list: split tokens
+        """
 
-        print("Extracted and processed files:", files_list)
\ No newline at end of file
+        matches = re.finditer('.+?(?:(?<=[a-z])(?=[A-Z])|(?<=[A-Z])(?=[A-Z][a-z])|$)', identifier)
+        return [m.group(0) for m in matches]
+    
+    def tokenize_text(text):
+        """
+        Tokenizes the text into individual words
+        Splits camelCase words
+        Removes cases
+
+        Args:
+            text (string): string to be tokenized
+
+        Returns:
+            list: tokens
+        """
+
+        tokens = []
+
+        # Tokenize the input text
+        for token in wordpunct_tokenize(text):
+            for word in Preprocessor.camel_case_split(token):
+                tokens.append(word)
+
+        # Remove short tokens
+        tokens = [token for token in tokens if len(token) > 2]
+
+        return tokens
+    
+    def remove_special_characters(text):
+        """
+        Removes special characters and punctuation from input text
+
+        Args:
+            text (string): string to have special chars removed 
+
+        Returns:
+            text (string): new string with special chars removed
+        """
+        
+        # Replace escape character with a ' '
+        text = text.replace("\n", " ")
+
+        # Replace special characters and numbers with a ' '
+        text = re.sub("[^A-Za-z\s]+", " ", text)
+        return text
+    
+    def preprocess_text(text, stop_words_path):
+        """
+        Preprocesses input text by
+            - Removing Numbers
+            - Removing special characters
+            - Removing punctuation
+            - Removing tokens of size 1-2
+            - Removing cases
+            - Removing inputted stop words
+
+        Args:
+            text (string): text to be preprocessed
+            stop_words (string): path to a stop words file
+
+        Returns:
+            string: preprocessed text
+        """
+
+        # Remove all special chars and punctuation from the text
+        text = Preprocessor.remove_special_characters(text)
+
+        # Tokenize the text
+        tokens = Preprocessor.tokenize_text(text)
+
+        # Read stop words from the input
+        with open(stop_words_path) as f:
```

Comment:
I know that we will be hard coding `stop_words_path`, but just for good measure we can put an error handler here and throw a `FileNotFoundError` if the path was hard coded incorrectly? It's not totally necessary but could save us the headache of debugging if something goes wrong later down the road.

Reply:
Fixed